corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -285,8 +285,18 @@ def default_value(self, user):
         return [Choice(SHOW_ALL_CHOICE, "[{}]".format(ugettext('Show All')))]
 
     def _locations_to_choices(self, locations):
+        cached_path_display = {}  # works best if self.order_by_hierarchy == True
+
         def display(loc):
-            return loc.get_path_display() if self.show_full_path else loc.display_name
+            if self.show_full_path:
+                if loc.parent_id in cached_path_display:
+                    path_display = '{}/{}'.format(cached_path_display[loc.parent_id], loc.name)
+                else:
+                    path_display = loc.get_path_display()
```

Comment:
Might be an overkill, but if this runs for all locations under a certain parent, it would still run `get_path_display` for all. 
Would it be better to actually add the parent full path if its missing by modifying something like
```
if loc.parent_id in cached_path_display:
   parent_full_path = cached_path_display[loc.parent_id]
else:
   cached_path_display[loc.parent_id] = parent_full_path = loc.parent.get_path_display()
path_display = '{}/{}'.format(parent_full_path, loc.name)
cached_path_display[loc.id] = path_display
```